src/lib/auth/auth.ts

```diff
@@ -0,0 +1,139 @@
+/* Facade for keycloak */
+import Keycloak, { KeycloakInitOptions, KeycloakInstance } from 'keycloak-js';
+
+export interface ResourceRole {
+  resource: string;
+  resourceId: string;
+  role: string;
+}
+
+export interface OrganizationRole {
+  orgNr: string;
+  role: string;
+}
+
+export interface AuthConfiguration {
+  oidcIssuer: string;
+  clientId: string;
+  redirectUri: string;
+  logoutRedirectUri: string;
+}
+
+export interface AuthInitOptions {
+  loginRequired: boolean;
+}
+
+export interface Auth {
+  init(options: AuthInitOptions): Promise<boolean>;
+
+  login(): Promise<void>;
+
+  logout(): Promise<void>;
+
+  isAuthenticated(): boolean;
+
+  getUser(): User;
+
+  getToken(): Promise<string>;
+
+  getAuthorizationHeader(): Promise<string>;
+
+  getAuthorities(): string;
+
+  getResourceRoles(): ResourceRole[];
+
+  hasResourceRole(r: ResourceRole): boolean;
+
+  hasOrganizationRole(organizationRole: OrganizationRole): boolean;
+
+  hasOrganizationReadPermission(orgNr: string): boolean;
+
+  hasOrganizationWritePermission(orgNr: string): boolean;
+
+  hasOrganizationAdminPermission(orgNr: string): boolean;
+
+  hasSystemAdminPermission(): boolean;
+}
+
+export interface User {
+  username: string;
+  name: string;
+}
+
+export const nullUser = { username: '', name: '' };
```

Comment:
not necessary

Reply:
this simplifies output type of getUser

Reply:
https://en.wikipedia.org/wiki/Null_object_pattern

Reply:
Does not make it a good pattern

Reply:
ok